sotu.md

```diff
@@ -1193,61 +1193,50 @@ overload, such as:
+In terms of testing, Haskell is a clear leader.
+It has both more practical, more robust and more advanced test automation than other languages.
 
-Haskell's `QuickCheck` is the gold standard which all other property-based
-testing libraries are measured against.  The reason `QuickCheck` works so
-smoothly in Haskell is due to Haskell's type class system and purity.  The type
-class system simplifies automatic generation of random data from the input type
-of the property test.  Purity means that any failing test result can be
+Haskell's property testing, which started with `QuickCheck`, is the gold
+standard which all other property-based testing implementations are measured
+against.  The reason property testing works so smoothly in Haskell is due to
+Haskell's strong static types and type class system.  The strong static types
+allow for automatic generation of what would otherwise be boiler plate;
+generators, shrinking, general properties.  The type class system simplifies
+how tests are defined.  Shrinking means that any failing test result can be
 automatically minimized by rerunning the check on smaller and smaller inputs
-until `QuickCheck` identifies the corner case that triggers the failure.
-
-Mocking is another area where Haskell shines because you can overload almost
-all built-in syntax, including:
-
-* `do` notation
-* `if` statements
-* numeric literals
-* string literals
-
-Haskell programmers overload this syntax (particularly `do` notation) to write
-code that looks like it is doing real work:
-
-```haskell
-example = do str <- readLine
-             putLine str
-```
+until the property testing framework identifies the corner case that triggers
+the failure.
 
-... and the code will actually evaluate to a pure syntax tree that you can use
-to mock in external inputs and outputs:
-
-```haskell
-example = ReadLine (\str -> PutStrLn str (Pure ()))
-```
-
-Haskell also supports most testing functionality that you expect from other
+Haskell supports all testing functionality that you expect from other
 languages, including:
 
 * standard package interfaces for testing
 * unit testing libraries
 * test result summaries and visualization
+* golden tests
+* mocking
+* integration tests with standard integrations like databases, web servers, ...
+
+On top of that, Haskell also supports more advanced testing features that are not common elsewhere:
+
+* Flakiness diagnostics
+* Deterministic randomness
+* Monadic test suite definition
+
 
 **Notable libraries:**
 
-* [`QuickCheck`](https://hackage.haskell.org/package/QuickCheck) - property-based testing
+- [`validity`](https://hackage.haskell.org/package/validity) - Property testing with smarter generators, Free generators, free shrinking, cheap properties
+- [`sydtest`](https://hackage.haskell.org/package/sydtest) - A modern testing framework for Haskell with good defaults and advanced testing features. It also has companion libraries for all major integration testing.
```

Comment:
`sydtest` appears to have a very restrictive [license](http://hackage.haskell.org/package/sydtest-0.1.0.0/src/LICENSE.md). Doesn't that make it unfit for recommendation in this guide? At the very least, a disclaimer should be added.

Reply:
`sydtest` is licenced such that I can maintain it in a sustainable way. This is not a limitation of Haskell the language and should not be a consideration here IMO